### PR TITLE
Sj completed date reset

### PR DESCRIPTION
**Pull request by Stuart-Johnson**

### CODE REVIEW

File: app/assets/javascripts/appointments.js.coffee [hunk 1]
```diff
@@ -79,6 +91,19 @@ $ ->
         url: "/multiple_procedures/reset_procedures.js"
         data: data
 
+  $(document).on 'click', '.uncomplete_visit', ->
+    appointment_id = $(this).parents('.row.appointment').data('id')
+    data = appointment: completed_date: null
+    $.ajax
+      type: 'PUT'
+      data: data
+      url: "/appointments/#{appointment_id}"
```
Comment (goleksiak) [inline app/assets/javascripts/appointments.js.coffee]: I think we should always include the :format in URLs. so url: "/appointments/#{appointment_id}.js"

File: app/assets/javascripts/appointments.js.coffee [hunk 2]
```diff
@@ -240,7 +265,7 @@ $ ->
   $(document).on 'change', '#appointment_content_indications', ->
     appointment_id = $(this).parents('.row.appointment').data('id')
     contents = $(this).val()
-    data = 'contents' : contents
+    data = appointment: 'contents' : contents
```
Comment (goleksiak) [inline app/assets/javascripts/appointments.js.coffee]: Is the key a sting for a reason? Should it be: data = appointment: contents: contents

File: app/views/appointments/_start_complete_buttons.html.haml [hunk 3]
```diff
@@ -16,6 +16,9 @@
       .form-group
         %label.col-xs-1.control-label#completed-date-label{for: 'completed_date'}= t(:appointment)[:completed_date]
         .col-xs-6= text_field_tag 'completed_date', '', class: 'form-control'
+        .col-xs-1#remove_completed_date_button
```
Comment (goleksiak) [inline app/views/appointments/_start_complete_buttons.html.haml]: Why set an id if never referenced?

Comment (Stuart-Johnson) [inline app/views/appointments/_start_complete_buttons.html.haml]: It is referenced in the CSS

File: app/assets/javascripts/appointments.js.coffee (hunk 2 above)
Comment (Stuart-Johnson) [inline app/assets/javascripts/appointments.js.coffee]: I'm not sure why it was like this myself, it's changed now.

**Pull request merged**